File: plugins/local/localdemo/localdemo.pl
#!/usr/bin/perl
use FindBin;
use lib "$FindBin::Bin/../lib/perl-lib/lib/perl5";
use lib "$FindBin::Bin/../lib";

use strict;
use Getopt::Long;

use Utils;

#帮助提示信息
###############################
sub usage {
    my $pname = $FindBin::Script;

    print("$pname --tinput <tinput> --tjson <tjson> --tselect <tselect> --tmultiselect <tmultiselect> --tpassword <tpassword> --tfile <tfile> --tnode <node id> --tdate <tdate> --ttime <ttime> --tdatetime <tdatetime>\n");
    exit(1);
}

#主程序
################################
sub main {
    Utils::setEnv();

    my ( $ishelp, $tinput, $tjson, $tselect, $tmultiselect, $tpassword, $tfile, $tnode, $tdate, $ttime, $tdatetime );

    GetOptions(
        'help'           => \$ishelp,
        'tinput:s'       => \$tinput,
        'tjson:s'        => \$tjson,
        'tselect:s'      => \$tselect,
        'tmultiselect:s' => \$tmultiselect,
        'tpassword:s'    => \$tpassword,
        'tfile:s'        => \$tfile,
        'tnode:s'        => \$tnode,
        'tdate:s'        => \$tdate,
        'ttime:s'        => \$ttime,
        'tdatetime:s'    => \$tdatetime
    );

    my $hasOptErr = 0;
    if ( not defined($tinput) or $tinput eq '' ) {
        $hasOptErr = 1;
    }
    if ( not defined($tselect) or $tselect eq '' ) {
        $hasOptErr = 1;
    }

    if ( $hasOptErr == 1 ) {
        usage();
    }

    print("=========GetOptions:\n");
    print("tinput:$tinput\n");
    print("tjson:$tjson\n");
    print("tselect:$tselect\n");
    print("tmultiselect:$tmultiselect\n");
    print("tpassword:$tpassword\n");
    print("tfile:$tfile\n");
    print("tnode:$tnode\n");
    print("tdate:$tdate\n");
    print("ttime:$ttime\n");
    print("tdatetime:$tdatetime\n");
    print("===================\n");

    #Do some job
    #process output
    my $out = {};

    $out->{outtext}     = "this is the text out value";
    $out->{outpassword} = "{RC4}xxxxxxxxxx";
    $out->{outfile}     = "this is the output file name";
    $out->{outjson}     = '{"key1":"value1", "key2":"value2"}';
    $out->{outcsv}      = q{"name","sex","age"\n"张三“,"男“,"30"\n"李四","女“,"35"};
    Utils::saveOutput($out);
}

#运行主程序
##############################
main();

File: plugins/local/localdemo/localdemo.py
#!/usr/bin/python

import os
import sys
import argparse
import Utils

binPaths = os.path.split(os.path.realpath(__file__))
libPath = os.path.realpath(binPaths[0]+'/../lib')
sys.path.append(libPath)


def usage():
    # 帮助提示信息
    pname = os.path.basename(__file__)
    print(pname + "--tinput <tinput> --tjson <tjson> --tselect <tselect> --tmultiselect <tmultiselect> --tpassword <tpassword> --tfile <tfile> --tnode <node id> --tdate <tdate> --ttime <ttime> --tdatetime <tdatetime>")


if __name__ == "__main__":
    # 参数处理
    parser = argparse.ArgumentParser()
    parser.add_argument('--tinput', default='', help='XXXXXX')
    parser.add_argument('--tjson', default='', help='XXXXXXX')
    parser.add_argument('--tselect', default='', help='XXXXXXX')
    parser.add_argument('--tmultiselect', default='', help='XXXXXXX')
    parser.add_argument('--tpassword', default='', help='XXXXXXX')
    parser.add_argument('--tfile', default='', help='XXXXXXX')
    parser.add_argument('--tnode', default='', help='XXXXXXX')
    parser.add_argument('--tdate', default='', help='XXXXXXX')
    parser.add_argument('--ttime', default='', help='XXXXXXX')
    parser.add_argument('--tdatetime', default='', help='XXXXXXX')

    args = parser.parse_args()

    # 主体处理逻辑
    print("Get options:============\n")
    print("tinput:" + args.tinput)
    print("tjson:" + args.tjson)
    print("tselect:" + args.tselect)
    print("tmultiselect:" + args.tmultiselect)
    print("tpassword:" + args.tpassword)
    print("tfile:" + args.tfile)
    print("tnode:" + args.tnode)
    print("tdate:" + args.tdate)
    print("ttime:" + args.ttime)
    print("tdatetime:" + args.tdatetime)

    print("Do some jobs=====\n")

    # 保存输出到json文件
    print("Save output to file\n")
    out = []
    out['outtext'] = "this is the text out value"
    out['outpassword'] = "{RC4}xxxxxxxxxx"
    out['outfile'] = "this is the output file name"
    out['outjson'] = '{"key1":"value1", "key2":"value2"}'
    out['outcsv'] = '"name","sex","age"\n"张三“,"男“,"30"\n"李四","女“,"35"}'
    Utils.saveOutput(out)
